Ginger/Ginger/UserControlsLib/UCTreeView/TreeView1.xaml.cs

```diff
@@ -108,22 +109,62 @@ private void xTreeViewTree_ItemSelected(object sender, EventArgs e)
         }
 
         private void xSearchTextBox_TextChanged(object sender, TextChangedEventArgs e)
-        {
+        {             
             if (xSearchTextBox.Text.Length > 0)
             {
                 xSearchNullText.Visibility= Visibility.Collapsed;
                 xSearchClearBtn.Visibility = Visibility.Visible;
             }
             else
-            {
-                xSearchNullText.Visibility = Visibility.Visible;
-                xSearchClearBtn.Visibility = Visibility.Collapsed;
+            {                                                
+                List<TreeViewItem> pathNodes = new List<TreeViewItem>();
+                if (xTreeViewTree.mlastSelectedTVI!=null)
+                {
+                    pathNodes = getSelecetdItemPathNodes(xTreeViewTree.mlastSelectedTVI);
+                }
+                ExpandCollapsePreviousState(xTreeViewTree.TreeItemsCollection, pathNodes);
```

Comment:
Rename the ExpandCollapsePreviousState, since we are not retaining the previous state now. We keep opened only the selected nodes.
Rename to -> CollapseUnselectedTreeNodes